src/goto-programs/abstract-interpretation/interval_domain.cpp

```diff
@@ -1019,69 +1018,68 @@ void interval_domaint::transform(
+  std::unordered_set<irep_idt, irep_id_hash> symbol_map;
+  for (const auto &myPair : a0)
+    symbol_map.insert(myPair.first);
+  for (const auto &myPair : a1)
+    symbol_map.insert(myPair.first);
+
+  // Here we apply the HULL operation (before, after)
+  for (const irep_idt &symbol : symbol_map)
   {
-    // search for the variable that needs to be merged
-    // containers have different sizes and ordering
-    const auto b_it = previous_map.find(new_it->first);
-    const auto f_it = fixpoint_map.find(new_it->first);
-    if (b_it == previous_map.end())
-    {
-      new_it = new_map.erase(new_it);
-      if (f_it != fixpoint_map.end())
-        fixpoint_map.erase(f_it);
-      result = true;
-    }
-    else
+    const auto previous_it = a0.find(symbol);
+    const auto next_it = a1.find(symbol);
+    const auto &update_it = previous_it;
+
+    // HULL(TOP, next_it) = TOP
+    if (previous_it == a0.end())
     {
-      auto previous = new_it->second; // [0,0] ... [0, +inf]
-      auto after = b_it->second;      // [1,100] ... [1, 100]
-      new_it->second.join(after);     // HULL // [0,100] ... [0, +inf]
-      // Did we reach a fixpoint?
-      if (new_it->second != previous)
+      // Narrowing
+      if (widening_narrowing)
       {
-        if (f_it != fixpoint_map.end())
-          f_it->second += 1;
-        else
-          fixpoint_map[new_it->first] = 0;
-
-        result = true;
-        // Try to extrapolate
-        if (
-          widening_extrapolate && fixpoint_map[new_it->first] > fixpoint_limit)
-        {
-          new_it->second = extrapolate_intervals(
-            previous,
-            new_it
-              ->second); // ([0,0], [0,100] -> [0,inf]) ... ([0,inf], [0,100] --> [0,inf])
-        }
+        // TODO: Narrowing needs more fixes
       }
+      continue;
+    }
 
-      else
-      {
-        // Found a fixpoint, we might try to narrow now!
-        if (widening_narrowing)
-        {
-          after = interpolate_intervals(
-            new_it->second,
-            b_it
-              ->second); // ([0,100], [1,100] --> [0,100] ... ([0,inf], [1,100] --> [0,100]))
-          result |= new_it->second != after;
-          new_it->second = after;
-        }
-      }
+    // HULL (previous_it, TOP) = TOP
+    if (next_it == a1.end())
+    {
+      result = true;
+      updated_map.erase(symbol);
+      continue;
+    }
 
-      new_it++;
+    const auto before = previous_it->second; // [0,0]
+    const auto &after = next_it->second;     // [1,100]
+    update_it->second.join(after);
+    const auto &joined = update_it->second;
+    if (before != joined)
+    {
+      result = true;
+      if (widening_extrapolate && should_extrapolate_instruction)
+        update_it->second = extrapolate_intervals(before, update_it->second);
+    }
+    else if (before != after && widening_narrowing)
+    {
+      // Narrowing
+      // TODO: Narrowing needs more fixes
```

Comment:
Which fixes?

Reply:
I described on item 2

Reply:
Once we merge your PR, nobody will check the PR description to know the remaining TODOs.